components/dashboard/src/App.tsx

```diff
@@ -40,6 +42,20 @@ function App() {
         return (<Login gitpodService={gitpodService} />)
     };
 
+    const hash = window.location.hash.replace(/^[#/]+/, '');
```

Comment:
Would be good to put the detection of a context URL into a function and give it name so the code is more readable.